src/main/java/org/fundacionjala/coding/carlos/eancode/EanValidator.java

```diff
@@ -0,0 +1,44 @@
+package org.fundacionjala.coding.carlos.eancode;
+
+/**
+ * This is a Kata of Codewars webpage.
+ */
+public final class EanValidator {
+    /**
+     * This was required for sonarcloud check fix.
+     */
+    private EanValidator() {
+    }
+
+    /**
+     * @param eanCode is the ean code.
+     * @return true or false
+     */
+    public static boolean validate(final String eanCode) {
```

Comment:
This may be refactored like this
```java
    public static boolean validate(final String eanCode) {
        int sum = 0;
        final int multiplier = 3;
        final int ten = 10;
        for (int i = 0; i < eanCode.length() - 1; i++) {
            int digit = Character.getNumericValue(eanCode.charAt(i));
            sum += i % 2 == 0 ? digit : multiplier * digit;
        }
        final int checksum = sum % ten == 0 ? 0 : ten - (sum % ten);
        return Character.getNumericValue(eanCode.charAt(eanCode.length() - 1)) == checksum;
    }
```